node/src/components/storage.rs

```diff
@@ -1424,6 +1438,21 @@ impl StorageInner {
         Ok(Some(block_header))
     }
 
+    fn get_block_header_by_height_restricted<Tx: Transaction>(
```

Comment:
We should really have a unit test for this new function.

Reply:
Added a new unit test for the newly added function